src/test/java/uk/gov/hmcts/reform/migration/CaseMigrationProcessorTest.java

```diff
@@ -0,0 +1,111 @@
+package uk.gov.hmcts.reform.migration;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.test.util.ReflectionTestUtils;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.domain.exception.CaseMigrationException;
+import uk.gov.hmcts.reform.migration.ccd.CoreCaseDataService;
+import uk.gov.hmcts.reform.migration.repository.ElasticSearchRepository;
+import uk.gov.hmcts.reform.migration.repository.IdamRepository;
+import uk.gov.hmcts.reform.migration.service.DataMigrationService;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+
+@RunWith(MockitoJUnitRunner.class)
+public class CaseMigrationProcessorTest {
+
+    private static final String USER_TOKEN = "Bearer eeeejjjttt";
+    private static final String EVENT_ID = "migrateCase";
+    private static final String EVENT_SUMMARY = "Migrate Case";
+    private static final String EVENT_DESCRIPTION = "Migrate Case";
+    private static final String CASE_TYPE = "Test_Case_Type";
+
+    @InjectMocks
+    private CaseMigrationProcessor caseMigrationProcessor;
+
+    @Mock
+    private CoreCaseDataService coreCaseDataService;
+
+    @Mock
+    private DataMigrationService dataMigrationService;
+
+    @Mock
+    private ElasticSearchRepository elasticSearchRepository;
+
+    @Mock
+    private IdamRepository idamRepository;
+
+    @Before
+    public void setUp() {
+        ReflectionTestUtils.setField(caseMigrationProcessor, "caseProcessLimit", 1);
+    }
+
+    @Test
```

Comment:
@kiran-yenigala-hmcts We can add a test case to cover invalid case type scenario